modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -415,7 +415,7 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
                         return prepareSvc.prepareAsync(result, ctx).thenApply(plan -> executePlan(session, txWrapper, ctx, plan));
                     }).whenComplete((res, ex) -> {
                         if (ex != null) {
-                            txWrapper.rollbackImplicit();
+                            txWrapper.rollback();
```

Comment:
Here, transaction will be rolled back if preparation step failed.
Is it expected?

Reply:
Yes. Invalid statements also cause transaction abort. (tested for Oracle, MySQL, and Postgres)